pytensor/tensor/slinalg.py

```diff
@@ -826,10 +818,93 @@ def solve_continuous_lyapunov(A: "TensorLike", Q: "TensorLike") -> TensorVariabl
     return _solve_continuous_lyapunov(A, Q)
 
 
+class SolveDiscreteARE(pt.Op):
+    __props__ = ("enforce_Q_symmetric",)
+
+    def __init__(self, enforce_Q_symmetric=False):
+        self.enforce_Q_symmetric = enforce_Q_symmetric
+
+    def make_node(self, A, B, Q, R):
+        A = as_tensor_variable(A)
+        B = as_tensor_variable(B)
+        Q = as_tensor_variable(Q)
+        R = as_tensor_variable(R)
+
+        out_dtype = pytensor.scalar.upcast(A.dtype, B.dtype, Q.dtype, R.dtype)
+        X = pytensor.tensor.matrix(dtype=out_dtype)
+
+        return pytensor.graph.basic.Apply(self, [A, B, Q, R], [X])
+
+    def perform(self, node, inputs, output_storage):
+        A, B, Q, R = inputs
+        X = output_storage[0]
+
+        if self.enforce_Q_symmetric:
+            Q = 0.5 * (Q + Q.T)
+        X[0] = scipy.linalg.solve_discrete_are(A, B, Q, R).astype(
```

Comment:
You should match the dtype of node.outputs[0].type instead. 

Imagine a user changes floatX after creating the node.